File: MixedProj/01.polyfit/main.py
import numpy as np
import time 
import datetime

cycles = 1000
name = '20'

#
def readBinFile( fileName ):
    file = open( fileName, 'rb' )
    data = np.fromfile ( file, dtype=np.double )	
    file.close()
    return data	
    


x = readBinFile('data/datax'+name+'.bin')
y = readBinFile('data/datay'+name+'.bin')

start = time.time()

for c in range( cycles ):
    a = np.polyfit(x,y,1)

end = time.time()


d = end-start

print ('# polyfit() X[',x.size,'] * ' , cycles ,' result: ', a )
print ('# time: ' , d  , '[sek.]' )
print ('')
print ('y[0]=' , d )
print ('')











start = time.time()

for c in range( cycles ):
    xsr=0
    ysr=0
    for i in range( len(x) ):
        xsr+=x[i]
        ysr+=y[i]          
    xsr=xsr/len(x)
    ysr=ysr/len(x)

    w0=0
    w1=0
    sumTop=0
    sumBottom=0
    for i in range( len(x) ):
        sumTop   +=(x[i]-xsr)*(y[i]-ysr)
        sumBottom+=(x[i]-xsr)*(x[i]-xsr)

    w1=sumTop/sumBottom
    w0=ysr - w1*xsr

end = time.time()
d = end-start


print ('# implementation X[',x.size,'] * ' , cycles ,' w0: ', w0 , ', w1: ' , w1  )
print ('# time: ' , d  , '[sek.]' )
print ('')
print ('y[1]=' , d )
print ('')

File: MixedProj/01.polyfit/Main.java
import java.io.File;
import java.io.FileInputStream;
import java.io.IOException;
import java.nio.ByteBuffer;
import java.time.Instant;
import java.time.temporal.ChronoUnit;

public class Main {

    private static  String name="20"; //
    private static Long cycles = 10000L;




    public static double[] loadBin( String filename ) throws IOException {
        double[] buf = null;
        int len;

        File f = new File(filename);
        len = (int) ( 0.125 * f.length());
        buf = new double[ len ];

        FileInputStream fis = new FileInputStream( filename );
        byte[] bB=new byte[8];
        byte[] bR=new byte[8];
        double val=0;
        for ( int i=0;i<len;i++ ) {

            fis.read(bB,0,8);
            bR[0]=bB[7];
            bR[1]=bB[6];
            bR[2]=bB[5];
            bR[3]=bB[4];
            bR[4]=bB[3];
            bR[5]=bB[2];
            bR[6]=bB[1];
            bR[7]=bB[0];
            val=ByteBuffer.wrap(bR).getDouble();
            buf[i]=val;

        }
        return buf;
    }




    public static void main(String[] args) throws IOException {
        double[] x = loadBin("data/datax"+name+".bin");
        double[] y = loadBin("data/datay"+name+".bin");

    double w1 = 0.0;
    double w0 = 0.0;

    

    Instant start = Instant.now();
    for ( int C=0; C<cycles; C++ ) {
        // -- start
        double xsr = 0.0;
        double ysr = 0.0;
        for (int i = 0; i < x.length; i++) {
            xsr += x[i];
            ysr += y[i];
        }
        xsr = xsr / x.length;
        ysr = ysr / y.length;

        w1 = 0.0;
        w0 = 0.0;
        double sumTop = 0.0;
        double sumBottom = 0.0;
        for (int i = 0; i < x.length; i++) {
            sumTop += ((x[i] - xsr) * (y[i] - ysr));
            sumBottom += ((x[i] - xsr) * (x[i] - xsr));
        }
        w1 = sumTop / sumBottom;
        w0 = ysr - w1 * xsr;
        //
    }
    Instant end = Instant.now();
    System.out.println( "# X[" + x.length + "] * " + cycles);
    System.out.println( "# time: " + (   ChronoUnit.MILLIS.between(start, end)/1000.0  ) + " [sek.],  result: [" + w1 + ", " + w0 + "]" );
    System.out.println( " " ); 
    System.out.println( "j[]=" + ChronoUnit.MILLIS.between(start, end)/1000.0 ); 
    }
}


















